src/cargo/ops/cargo_read_manifest.rs

```diff
@@ -183,11 +183,13 @@ fn read_nested_packages(
             v.insert(pkg);
         }
         Entry::Occupied(_) => {
-            let _ = config.shell().warn(format!(
-                "skipping duplicate package `{}` found at `{}`",
-                pkg.name(),
-                path.to_string_lossy()
-            ));
+            if pkg.publish().is_none() {
+                let _ = config.shell().warn(format!(
+                    "skipping duplicate package `{}` found at `{}`",
+                    pkg.name(),
+                    path.to_string_lossy()
+                ));
```

Comment:
Could we add a comment clarifying why we did this (see my explanation in the other comment)?

Reply:
Sure